### PR TITLE
completions API change + tox

### PR DESCRIPTION
**Pull request by sdorman-lambda**
## Primary change
Updates `main.py` to use the `chat/completions` API as opposed to the legacy `completions`
* Docs here: https://platform.openai.com/docs/api-reference/chat

The changes to `main.py` look large, but it's mostly linting and formatting. The only functional changes are noted in the comments.

## QoL changes
* Adds tox.ini with `black` and `flake8` for python formatting and linting
  * Running `tox` from repo root will format `*.py` files and check lint
* Adds alternative to manual venv creation by adding a `tox -e venv` option
* Updates README 

## Notes
Check out the commit message to see what AI came up with 😄  I used the `gpt-4o-mini` model.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -49,33 +63,35 @@ def main():
...
 
     # Run the model with the prompt
-    response = client.completions.create(
-      prompt=full_prompt,
-      temperature=0,
-      model=MODEL,
+    response = client.chat.completions.create(
+        messages=[{"role": "user", "content": full_prompt}],
+        model=MODEL,
     )
```
Comment (sdorman-lambda) [inline main.py:87]: This is the change to `chat/completions` API

File: main.py [hunk 2]
```diff
@@ -49,33 +63,35 @@ def main():
...
 
     # We are ready to provide the commit message
-    spinner.succeed('Done')
+    spinner.succeed("Done")
 
     # Commit with the generated message
-    subprocess.run(['git', 'commit', '-e', '-m', response.choices[0].text])
+    subprocess.run(["git", "commit", "-e", "-m", response.choices[0].message.content])
```
Comment (sdorman-lambda) [inline main.py:93]: The return from `chat/completions` is a bit different from `completions`, so this gets us the goods

**Review by sdorman-lambda — COMMENTED** [review]
These are the only functional changes to the code

**Review by lambdajosh — APPROVED** [review]
Never worked with tox before, was really easy to setup and use.
I'm going to start using this for my python virtual environments!

**Pull request merged**